src/main/kotlin/solve/scene/controller/SceneController.kt

```diff
@@ -4,31 +4,66 @@ import javafx.beans.property.SimpleDoubleProperty
 import javafx.beans.property.SimpleObjectProperty
 import solve.scene.model.Scene
 import solve.scene.view.SceneView
-import solve.utils.ceilToInt
 import solve.utils.structures.Point as DoublePoint
 import tornadofx.Controller
+import tornadofx.onChange
 import kotlin.math.max
 import kotlin.math.min
-import kotlin.math.sqrt
 
 class SceneController : Controller() {
```

Comment:
Grid settings should be related to a concrete project, when new project is imported it should start with default settings (like now in master). Else we have this terrible behavior 
![s1tnfn0Nme](https://user-images.githubusercontent.com/55746901/229488620-cba4e606-3136-4333-8505-041ef1e7cf5b.gif)